### PR TITLE
SALTO-2392 resolve account specific values in workflows

### PR DESCRIPTION
**Pull request by dantal4**
Use queried internal ids maps from SuiteQL and the "record" SuiteApp operation to get the value behind ASV in workflows.

---

_Additional context for reviewer_

TODO:
- [x] add tests

---
_Release Notes_: 
Netsuite Adapter:
- resolve account specific values in workflows (behind config value `fetch.resolveAccountSpecificValues`)

---
_User Notifications_: 
None

### CODE REVIEW

File: packages/netsuite-adapter/src/change_validators/workflow_account_specific_values.ts [hunk 1]
```diff
@@ -60,56 +60,111 @@ const isNestedValueChanged = (
...
+  await handleWorkflowAccountSpecificValuesOnDeploy(
+    instances,
+    elementsSource,
+    {
+      onMissingInternalId: ({ path, value, field, name }) => {
+        warnings.push({
+          elemID: path,
+          severity: 'Warning',
+          message: 'Could not identify value in workflow',
+          detailedMessage: `Could not find object "${name}" for field "${field}". Setting ACCOUNT_SPECIFIC_VALUE instead.`,
```
Comment (dantal4) [inline packages/netsuite-adapter/src/change_validators/workflow_account_specific_values.ts:122]: @liorn can you help rephrase this warning?

File: packages/netsuite-adapter/src/change_validators/workflow_account_specific_values.ts [hunk 2]
```diff
@@ -60,56 +60,111 @@ const isNestedValueChanged = (
...
+      },
+      onSingleInternalId: ({ value, field, internalId }) => {
+        value[field] = internalId
+      },
+      onMultipleInternalIds: ({ path, value, field, name, internalIds: [internalId] }) => {
+        warnings.push({
+          elemID: path,
+          severity: 'Warning',
+          message: 'Multiple objects with the same name',
+          detailedMessage: `There are multiple objects with the name "${name}". Using the first one (internal id: ${internalId}).`,
```
Comment (dantal4) [inline packages/netsuite-adapter/src/change_validators/workflow_account_specific_values.ts:134]: @liorn can you help rephrase this warning?

File: packages/netsuite-adapter/src/filters/workflow_account_specific_values.ts [hunk 3]
```diff
@@ -0,0 +1,768 @@
...
+    },
+  })
+
+  return resolveWarnings
+}
+
+export const handleWorkflowAccountSpecificValuesOnDeploy = async (
+  instances: InstanceElement[],
+  elementsSource: ReadOnlyElementsSource,
+): Promise<ChangeError[]> => {
```
Comment (omrilit) [inline packages/netsuite-adapter/src/filters/workflow_account_specific_values.ts:687]: I think that the naming here is not indicative enough.
Maybe resolveWorkflowsAccountSpecificValues would be better? WDYT?
```suggestion
export const resolveWorkflowsAccountSpecificValues = async (
  workflowInstances: InstanceElement[],
  elementsSource: ReadOnlyElementsSource,
): Promise<ChangeError[]> => {
```

**Review by omrilit — COMMENTED** [review]
Completed reviewing most of the code, will continue tomorrow or in the day after, SUPER impressive meanwhile!

File: packages/netsuite-adapter/src/filters/workflow_account_specific_values.ts [hunk 4]
```diff
@@ -0,0 +1,772 @@
...
+    .toArray()
+
+  return _(suiteQLTableInstances)
+    .keyBy(instance => instance.elemID.name)
+    .mapValues(instance => instance.value[INTERNAL_IDS_MAP] as InternalIdsMap)
+    .mapValues(
+      internalIdsMap => _(internalIdsMap)
+        .entries()
+        .groupBy(row => row[1].name)
+        .mapValues(rows => rows.map(row => row[0]))
```
Comment (omrilit) [inline packages/netsuite-adapter/src/filters/workflow_account_specific_values.ts:610]: I think this is better in terms of readibility
```suggestion
        .groupBy((_key, value) => value.name)
        .mapValues(rows => rows.map((key, _value) => key))
```

**Review by omrilit — APPROVED** [review]
💯

**Pull request merged**